torch/csrc/distributed/rpc/rpc_agent.cpp

```diff
@@ -6,16 +6,134 @@ namespace rpc {
 
 constexpr size_t WorkerInfo::MAX_NAME_LEN;
 
+// TODO(osalpekar): Tune RetryDuration to be something in seconds per
+// team recommendations. Kept at 100ms for the sake of Milestone 1.
+const auto kRpcRetryDuration = std::chrono::milliseconds(100);
+// kBackoff is the multiplier for the exponential backoff algorithm.
+const float kBackoff = 1.2;
+
 RpcAgent::RpcAgent(
     WorkerInfo workerId,
     std::unique_ptr<RequestCallback> cb,
     std::chrono::milliseconds rpcTimeout)
     : workerInfo_(std::move(workerId)),
       cb_(std::move(cb)),
       rpcTimeout_(rpcTimeout),
-      profilingEnabled_(false) {}
+      profilingEnabled_(false) {
+  {
+    std::lock_guard<std::mutex> lock(rpcRetryMutex_);
+    rpcAgentRunning_.store(true);
```

Comment:
Can we dedupe this with the `std::atomic<bool> rpcRunning_{false};` in processGroupAgent?

Reply:
Yep, that's the plan. I'll be making `processGroupAgent`-related changes in a future PR per the design doc

Reply:
That makes sense. Although, it may be bug prone or at least confusing to see 2 sources of truth for if the RPC agent is running or not, before the PG agent changes get landed, hence why it may be useful to do the deduping here itself. I don't have too much of a preference either way.

Reply:
Cool, I'll add a comment to the atomic variable in PGA saying it's likely to be replaced with the one in the parent class - hopefully that clears some confusion in the meantime!  I think let's go for doing PGA-stuff in a future PR, since we're also thinking about moving the RPC Timeout Watchdog thread to `RpcAgent` and need to make sure this all plays well with the existing `start()` and `shutdown()` functions. This would also make for more functionally-separated PR's. Open to alternate suggestions though!